src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Windows.vb

```diff
@@ -0,0 +1,310 @@
+            Dim FinalSourcePath As String = FullSourcePath
+            If TargetType = FileOrDirectory.Directory Then
+                ' Shell behavior: If target does not exist, create target and copy / move source CONTENT into target.
+                '                 If target exists, copy / move source into target.
+                ' To have our behavior:
+                '   If target does not exist, create target parent (or shell will throw) and call ShellCopyOrMove.
+                '   If target exists, attach "\*" to FullSourcePath and call ShellCopyOrMove.
+                ' In case of Move, since moving the directory, just create the target parent.
+                If IO.Directory.Exists(FullTargetPath) Then
+                    FinalSourcePath = IO.Path.Combine(FullSourcePath, "*")
+                Else
+                    IO.Directory.CreateDirectory(GetParentPath(FullTargetPath))
+                End If
+            End If
+
+            'Call into ShellFileOperation.
+            ShellFileOperation(OperationType, OperationFlags, FinalSourcePath, FullTargetPath, OnUserCancel, TargetType)
+
+            ' *** Special action for Directory only. ***
+            ' In case target does exist, and it's a move, we actually move content and leave the source directory.
+            ' Clean up here.
+            If Operation = CopyOrMove.Move AndAlso TargetType = FileOrDirectory.Directory Then
+                If IO.Directory.Exists(FullSourcePath) Then
+                    If IO.Directory.GetDirectories(FullSourcePath).Length = 0 _
+                        AndAlso IO.Directory.GetFiles(FullSourcePath).Length = 0 Then
+                        IO.Directory.Delete(FullSourcePath, recursive:=False)
+                    End If
+                End If
+            End If
+
+        End Sub
+
+        ''' <summary>
+        ''' Sets relevant flags on the SHFILEOPSTRUCT and calls into SHFileOperation to delete file / directory.
+        ''' </summary>
+        ''' <param name="FullPath">Full path to the file / directory.</param>
+        ''' <param name="ShowUI">ShowDialogs to display progress and confirmation dialogs. Otherwise HideDialogs.</param>
+        ''' <param name="recycle">SendToRecycleBin to delete to Recycle Bin. Otherwise DeletePermanently.</param>
+        ''' <param name="OnUserCancel">Throw exception or not if the operation was canceled (by user or errors in the system).</param>
+        ''' <remarks>
+        ''' We don't need to consider Recursive flag here since we already verify that in DeleteDirectory.
+        ''' </remarks>
+        Private Shared Sub ShellDelete(ByVal FullPath As String,
+            ByVal ShowUI As UIOptionInternal, ByVal recycle As RecycleOption, ByVal OnUserCancel As UICancelOption, ByVal FileOrDirectory As FileOrDirectory)
+
+            Debug.Assert(FullPath.Length <> 0 AndAlso IO.Path.IsPathRooted(FullPath), "FullPath must be a full path!!!")
```

Comment:
>FullPath.Length <> 0 AndAlso [](start = 25, length = 28)

`FullPath <> "" And`

Reply:
We should leave the existing code as is since there's no reason to change it.

---
In reply to: [227927205](https://github.com/dotnet/corefx/pull/32668#discussion_r227927205) [](ancestors = 227927205)